### PR TITLE
Add Support for connection with Proxy in Databricks provider

### PR DESCRIPTION
**Pull request by microcassidy**
PR is in reference to #32576. Allows for proxied connections to azure databricks

### CODE REVIEW

**Comment by eladkal** [comment]
cc @alexott

File: airflow/providers/databricks/hooks/databricks_base.py [hunk 1]
```diff
@@ -248,9 +248,11 @@ def _get_aad_token(self, resource: str) -> str:
                         azure_ad_endpoint = self.databricks_conn.extra_dejson.get(
                             "azure_ad_endpoint", AZURE_DEFAULT_AD_ENDPOINT
                         )
+                        azure_databricks_proxy = self.databricks_conn.extra_dejson.get("databricks_proxy")
```
Comment (alexott) [inline airflow/providers/databricks/hooks/databricks_base.py]: I would go with simpler name - `proxies`, or something like that.

File: airflow/providers/databricks/hooks/databricks_base.py [hunk 2]
```diff
@@ -248,9 +248,11 @@ def _get_aad_token(self, resource: str) -> str:
...
                         resp = requests.post(
                             AZURE_TOKEN_SERVICE_URL.format(azure_ad_endpoint, tenant_id),
                             data=data,
+                            proxies=azure_databricks_proxy,
```
Comment (alexott) [inline airflow/providers/databricks/hooks/databricks_base.py]: If we're doing that for AAD tokens, shouldn't we do it for "normal" HTTP requests for APIs themselves? Or we may need to specify which `requests` calls should be proxied.

**Review by alexott — CHANGES_REQUESTED** [review]
See comments. Also

* We need documentation for that feature.
* Also add support for it in the async part of the operators.

File: airflow/providers/databricks/hooks/databricks_base.py (hunk 1 above)
Comment (microcassidy) [reply airflow/providers/databricks/hooks/databricks_base.py]: Thanks for taking the time to review. I have gone with your suggestions

File: airflow/providers/databricks/hooks/databricks_base.py (hunk 2 above)
Comment (microcassidy) [reply airflow/providers/databricks/hooks/databricks_base.py]: You're probably right. Which direction do you think is the most sensible?

**Comment by xuky76** [comment]
@alexott I think I missed adding "proxies" into [extra_parameters](https://github.com/xuky76/airflow/blob/9dfc51a8cf5027b8180e9d4e78febb397aca4ce1/airflow/providers/databricks/hooks/databricks_base.py#L91)

If the other requests are requiring proxies, should I move the proxies parameter into a `@cached_property`block?

Sorry for all the questions - I'd just rather do it properly, rather than again.

**Comment by alexott** [comment]
@xuky76 I agree about "making it right". Cached property could help to simplify access to the data, but even before we need to decide about the core functionality.  There are different things that needs to be taken into consideration:

1. Access to AAD authentication REST APIs - the piece that you added
2. Access to the REST API - here the things could be different depending if traffic is going via public REST API or via private link. For the latter case we don't need proxy config, so we may need to have an additional configuration specifying if we should use proxy for it or not.
3. Many proxies require authentication - we need to understand how it could be configured, etc.
4. Similar proxy configuration may need to be added to the SQL operators as well.

**Comment by xuky76** [comment]
> @xuky76 I agree about "making it right". Cached property could help to simplify access to the data, but even before we need to decide about the core functionality. There are different things that needs to be taken into consideration:
> 
>     1. Access to AAD authentication REST APIs - the piece that you added
> 
>     2. Access to the REST API - here the things could be different depending if traffic is going via public REST API or via private link. For the latter case we don't need proxy config, so we may need to have an additional configuration specifying if we should use proxy for it or not.
> 
>     3. Many proxies require authentication - we need to understand how it could be configured, etc.
> 
>     4. Similar proxy configuration may need to be added to the SQL operators as well.

It seems like the configuration options are quite open ended...perhaps the extra proxies field could be more nested and  the AAD component just get accessed with an additional get call in the requests block? at least that way we can leave room for other proxy configurations in the future.

I don't understand what you mean by 4. in relation to this issue. Should we just resolve the AAD issue and then explore this in a separate issue or are they related and inseparable?

**Comment by alexott** [comment]
Regarding 4th item - I meant that we may need to track it separately & also fix.

**Pull request closed without merging**